Do not allocate temporary data in PortalContext

akuzm opened the pull request:
The hyperspace point corresponding to the inserted row should be
allocated in the per-tuple context.

Fixes: https://github.com/timescale/timescaledb/issues/4576 (Fabrizio repoted that this fix largely helps).

fabriziomello commented:
@akuzm I repeat the tests locally using this PR and we still leak somewhere, after some minutes running the COPY the backend memory went over 4GB

```bash
➜ ps faux | grep postgres | grep COPY
fabrizio   67960 92.7  6.0 4057900 3960964 ?     RNs  14:52   9:25      \_ postgres: fabrizio fabrizio [local] COPY
```

akuzm commented:
Somehow it only starts to leak for real after 30 minutes for me. I fixed some other leaks, fixed my tools that I used to find leaks, and also added a test for memory usage in PortalContext. Should be better now.

mfundul commented on src/hypertable.c:
```diff
@@ -1048,7 +1048,7 @@ hypertable_chunk_store_add(const Hypertable *h, const Chunk *input_chunk)
 						  cached_chunk->cube,
 						  cached_chunk,
-						  /* object_free = */ NULL);
+						  (void (*)(void *)) ts_chunk_free);
```
That's a bit of a risky cast. Why not do it inside the function like in `/src/nodes/chunk_dispatch.c`:
```
static void
destroy_chunk_insert_state(void *cis)
{
	ts_chunk_insert_state_destroy((ChunkInsertState *) cis);
}
```

akuzm replied:
You mean, in case the signature changes? Sure, I can do it.

mfundul replied:
Technically it is considered undefined behaviour. I'm sure it works in practice e.g. for gcc.

fabriziomello commented:
> Somehow it only starts to leak for real after 30 minutes for me. I fixed some other leaks, fixed my tools that I used to find leaks, and also added a test for memory usage in PortalContext. Should be better now.

@akuzm In my laptop start leak immediately ... :-( 

Much better now with the current version.

```bash
➜ ps faux | grep postgres | grep COPY
fabrizio  696080 99.6  0.2 250356 165704 ?       RNs  18:24   6:28      \_ postgres: fabrizio fabrizio [local] COPY

➜ ps faux | grep postgres | grep COPY
fabrizio  696080 99.8  0.2 254560 168372 ?       RNs  18:24  10:58      \_ postgres: fabrizio fabrizio [local] COPY

➜ ps faux | grep postgres | grep COPY
fabrizio  696080 99.7  0.2 257068 169708 ?       RNs  18:24  15:24      \_ postgres: fabrizio fabrizio [local] COPY

➜ ps faux | grep postgres | grep COPY
fabrizio  696080 99.1  0.2 287640 196272 ?       RNs  18:24  20:35      \_ postgres: fabrizio fabrizio [local] COPY
```

Still increasing a bit over the time but orders of magnitude less than `main` branch.

Ps: and now my cpu fan is saying "thank you AlexK"
